src/fungible_token.rs

```diff
@@ -57,6 +57,12 @@ impl FungibleToken {
                 .total_supply_near
                 .checked_add(amount)
                 .expect("ERR_TOTAL_SUPPLY_OVERFLOW");
+            // Note: Reusing total supply for a sum of ETH and NEAR is very weird. It doesn't give
+            // mean anything reasonable. I guess the goal is to just report changing numbers for the
+            // total supply when either ETH or NEAR balances are changing. In this case it's probably
+            // okay to report only one of them, e.g. ETH total_supply and not report NEAR total
+            // supply through `ft_total_supply`.
+            // TODO: Remove total_supply for NEAR
```

Comment:
I was really confused about this as well when I first read the code (see https://github.com/aurora-is-near/aurora-engine/pull/24#discussion_r616178696 ). But I think all the units are the same in the struct (all amounts are relating to $ETH) and the separation is about how much $ETH is owned by Aurora EVM addresses vs NEAR accounts. (@mrLSD please correct me if I'm wrong here).

That said, if @evgenykuzyakov made the same mistake then maybe we should really think about trying to express this more clearly in the code. A simple thing could be making a type alias like `type NEth = u128`. More comments describing the flow at a high level could also help.

Reply:
Yes, we can simplify that. But do we really need that? It'll take 10 mins to remove that.